### ISSUE TITLE
Custom Grammar Support

### ISSUE THREAD

**Issue by joeladria** [alphacep/vosk-asterisk#32]
Any chance grammar loading will be implemented?

https://github.com/alphacep/vosk-asterisk/blob/f5f800cba19cf41ec649d4888d6778f3e4b7aada/res-speech-vosk/res_speech_vosk.c#L121

**Comment by nshmyrev** [comment]
It should be very easy to send a grammar message to configure the recognizer! Please submit a patch or let us know if you need help implementing it!

**Comment by joeladria** [comment]
I wish I could help out but my C experience isn't quite at that level unfortunately! I've only found the grammar implementation examples for Python so far... https://github.com/alphacep/vosk-api/blob/master/python/example/test_words.py

**Comment by joeladria** [comment]
Looking for some feedback, I tried my hand at making it work.

Asterisk res_speech specifies the loading of [grammar files](https://www.w3.org/TR/speech-grammar/) which are then named at load time and then unloaded. Does vosk support or implement these grammar files at all?

In my implementation I've simply used the "name" string as input with the array of words to pass to the model in JSON format. Works nicely, but a bit of a hack. What do you think?

https://github.com/joeladria/vosk-asterisk/commit/4d7e05b7288462ab53d1d0024680d561a09b020e

**Comment by nshmyrev** [comment]
Yup, something like this should work. It needs a bit more love though for full integration.
